app/models/bike.rb

```diff
@@ -497,6 +497,27 @@ def external_image_urls
     b_params.map { |bp| bp.external_image_urls }.flatten.reject(&:blank?).uniq
   end
 
+  def alert_image_url
+    return if current_stolen_record.blank?
+
+    generate_alert_image
+    current_stolen_record.alert_image_url
+  end
+
+  def generate_alert_image
+    return if current_stolen_record.blank?
+    return if current_stolen_record.alert_image.present?
```

Comment:
How do we trigger regenerating if there is a new public image first?

Reply:
`generate_alert_image` now uses the most recently added bike image and re-generates the alert image only if the bike image has changed (using the filename equivalence as a heuristic)

```rb
# app/models/bike.rb L530-534 (55c09ed7)

def alert_image_url
  return if current_stolen_record.blank? || public_images.none?
  current_stolen_record.generate_alert_image
  current_stolen_record.alert_image_url
end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/55c09ed7/app/models/bike.rb#L530-L534)]</sup>

```rb
# app/models/stolen_record.rb L262-281 (a0dc6a32)

# Generate the "premium alert image"
# (the most recently created bike image placed on a branded template)
#
# The URL is available immediately, processing is performed in the background.
# If processing fails, the unprocessed image will be loaded.
#
# TODO: Allow selection of which bike image to use for the alert image
def generate_alert_image
  new_image = bike.reload.public_images.order(:created_at).last&.image
  return if new_image.blank?

  if alert_image.present?
    new_file = File.basename(new_image.path, ".*")
    cur_file = File.basename(alert_image.path, ".*")
    return if "#{new_file}-alert" == cur_file
  end

  alert_image.remove!
  update(alert_image: new_image)
end
```
<sup>[[source](https://github.com/bikeindex/bike_index/blob/a0dc6a32/app/models/stolen_record.rb#L262-L281)]</sup>